pkg/controller/atlasdatabaseuser/atlasdatabaseuser_controller.go

```diff
@@ -194,6 +204,32 @@ func (r *AtlasDatabaseUserReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	return result.ReconcileResult(), nil
 }
 
+func (r *AtlasDatabaseUserReconciler) handleFeatureFlags(dbuser *mdbv1.AtlasDatabaseUser) error {
+	if r.FeatureFlags == nil {
+		return nil
+	}
+
+	err := handleOIDCPreview(r.FeatureFlags, dbuser)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO: Remove after the OIDC feature becomes stable
+func handleOIDCPreview(featureFlags *featureflags.FeatureFlags, dbuser *mdbv1.AtlasDatabaseUser) error {
+	if dbuser == nil {
+		return nil
+	}
+
+	if !featureFlags.IsFeaturePresent(featureflags.FeatureOIDC) && dbuser.Spec.OIDCAuthType != "" {
```

Comment:
suggestion: let's handle feature flags centrally, more near to `main` and just dependency-inject a `oidcEnabled bool` field into the reconciler struct to keep logic in one place.

Reply:
(this would reduce cross-package dependencies as well)

Reply:
Good point! Fixed